qemu/tests/qemu_guest_agent_update.py

```diff
@@ -105,6 +106,25 @@ def _qga_install():
         vm = self.env.get_vm(params["main_vm"])
         session = self._get_session(params, vm)
 
+        if params.get("driver_uninstall", "no") == "yes":
+            error_context.context("Uninstall vioser driver in guest.",
+                                  logging.info)
+            device_name = params["device_name"]
+            driver_name = params["driver_name"]
+            inf_names_get_cmd = wmic.make_query("path win32_pnpsigneddriver",
+                                                "DeviceName like '%s'" %
+                                                device_name,
+                                                props=["InfName"],
+                                                get_swch=wmic.FMT_TYPE_LIST)
+            inf_names = wmic.parse_list(session.cmd(inf_names_get_cmd,
+                                                    timeout=360))
+            for inf_name in inf_names:
+                uninst_store_cmd = "pnputil /f /d %s" % inf_name
+                s, o = session.cmd_status_output(uninst_store_cmd, 360)
+                if s:
+                    test.error("Failed to uninstall driver '%s' from store, "
+                               "details:\n%s" % (driver_name, o))
+
```

Comment:
@xiagao In my understanding, This part code only make sure uninstall vioser device driver.  no found how to upgrade qemu-ga-win package.

Reply:
The original qemu_guest_agent_update.py mechanized the upgrade steps already.

Reply:
I am not familiar windows command.  only review code's function.  It can make sure vioser driver is uninstalled.

Acked: Xiangchun Fu xfu@redhat.com